lib/Sema/CSGen.cpp

```diff
@@ -2766,9 +2779,12 @@ namespace {
           // of is-patterns applied to an irrefutable pattern.
           pattern = pattern->getSemanticsProvidingPattern();
           while (auto isp = dyn_cast<IsPattern>(pattern)) {
-            Type castType = TypeResolution::forContextual(
-                                CS.DC, TypeResolverContext::InExpression)
-                                .resolveType(isp->getCastTypeRepr());
+            const Type castType = TypeResolution::forContextual(
+                                      CS.DC, TypeResolverContext::InExpression,
+                                      // FIXME: Should we really unconditionally
```

Comment:
That would be a source break. I think it's fine to have unbound generics here if the generic arguments can be inferred from context. Eg, `let foo: [Int] = blah as Array`?

Reply:
Ah, I should have removed this comment (to yell at an unbound generic type  in 
`} catch is Foo {` is the right thing to do).